ONSdigital/rm-action-service
# Remove bi/create multiple requests from actions

# Motivation and Context
When we remove `BI` cases we will only have one `B` case in the action service, event for multiple respondents. We still need to generate the correct actions/notifications from this single case

# What has changed

1. Replaced SQL query for creating actions with code.
2. Made small changes to action distribution process, remove references to parent/child parties and replace with business/respondent parties in relevant places
4. Updated `CaseNotificationService` to handle `ACTIONPLAN_CHANGED` notification types
5. Refactored execution of action plans to require fewer external calls
6. Updated some logging

# How to test?
All tests should pass on `mvn clean install`

Test multiple requests can be created from one action
1. Have two businesses registered for one case (business/survey combination)
2. Case should be on a respondent action plan (formerly BI case action plan)
3. Trigger an action rule for the respondent action plan
4. Two action requests should be created adding messages to the notifygateway queue

Once other BI case work is ready try acceptance tests with all relevant PR's

# Links
[Trello](https://trello.com/c/4Sb2MBAb/302-create-multiple-actions-emails-from-single-b-case-actions-xl)
